### DISCUSSION TITLE
Code cleanup

### DISCUSSION THREAD

**Discussion by YaSuenag** [Green-Software-Foundation/discussions#62] [Carbon Aware SDK]
Do you have any plan to refine the SDK code?

I executed "Analyze and Code Cleanup" with all fixers on VS 2022, then a lot of lines were fixed. All of them don't need to be fixed of course, but some of them would be useful for code cleanup. So I think it is better to create plans/issues/PRs for them (targeted to next release (v1.2)?)

Maybe we can use [dotnet/code-analysis](https://github.com/dotnet/code-analysis) action for this purpose? I'm happy we can get the help about this from MS folks.

**Comment by Willmish** [comment]
I think this could be useful refactor, as mentioned by @YaSuenag - maybe v1.2? what do you think @vaughanknight , @bderusha , @danuw , @JenMadiedo ?

Are you able to post the report here @YaSuenag for us to see? otherwise I can try and run it myself to repost here

**Comment by YaSuenag** [comment]
I saw 201 messages on entire of solution with Visual Studio 2022 (v17.7.1).
For example, we can see 33 messages on CarbonAware.LocationSources project in below (my VS is configured in Japanese locale, sorry 😐 )

重大度レベル	コード	説明	プロジェクト	ファイル	行	抑制状態
メッセージ	CA2254	ログ メッセージ テンプレートは、'LoggerExtensions.LogInformation(ILogger, string?, params object?[])' への呼び出しによって異なるべきではありません。	CarbonAware.LocationSources	D:\github-forked\carbon-aware-sdk\src\CarbonAware.LocationSources\src\LocationSource.cs	91	アクティブ
メッセージ	CA1822	メンバー 'BuildKey' はインスタンス データにアクセスしないため、static にマークできます	CarbonAware.LocationSources	D:\github-forked\carbon-aware-sdk\src\CarbonAware.LocationSources\src\LocationSource.cs	76	アクティブ
メッセージ	CA2254	ログ メッセージ テンプレートは、'LoggerExtensions.LogDebug(ILogger, string?, params object?[])' への呼び出しによって異なるべきではありません。	CarbonAware.LocationSources	D:\github-forked\carbon-aware-sdk\src\CarbonAware.LocationSources\src\LocationSource.cs	94	アクティブ
メッセージ	CA2254	ログ メッセージ テンプレートは、'LoggerExtensions.LogWarning(ILogger, string?, params object?[])' への呼び出しによって異なるべきではありません。	CarbonAware.LocationSources	D:\github-forked\carbon-aware-sdk\src\CarbonAware.LocationSources\src\LocationSource.cs	108	アクティブ
メッセージ	CA2254	ログ メッセージ テンプレートは、'LoggerExtensions.LogInformation(ILogger, string?, params object?[])' への呼び出しによって異なるべきではありません。	CarbonAware.LocationSources	D:\github-forked\carbon-aware-sdk\src\CarbonAware.LocationSources\src\LocationSource.cs	111	アクティブ
